namerd/storage/consul/src/main/scala/io/buoyant/namerd/storage/consul/ConsulDtabStore.scala

```diff
@@ -0,0 +1,102 @@
+package io.buoyant.namerd
+package storage.consul
+
+import com.twitter.finagle.{Dtab, Path}
+import com.twitter.io.Buf
+import com.twitter.util._
+import io.buoyant.consul.NotFound
+import io.buoyant.consul.v1._
+
+import io.buoyant.namerd.DtabStore.{DtabNamespaceAlreadyExistsException, DtabNamespaceDoesNotExistException}
+
+class ConsulDtabStore(api: KvApi, root: Path) extends DtabStore {
+  import DtabStore.{DtabVersionMismatchException, Version}
+
+  override def list(): Activity[Set[Ns]] = {
+    def namespace(key: String): Ns = key.stripPrefix("/").stripSuffix("/").substring(root.show.length)
+
+    val run = Var.async[Activity.State[Set[Ns]]](Activity.Pending) { updates =>
+      @volatile var running = true
+
+      def cycle(index: Option[String]): Future[Unit] =
+        if (running)
+          api.list(s"${root.show}/", blockingIndex = index) transform {
+            case Return(result) =>
+              val namespaces = result.value.map(namespace).toSet
+              updates() = Activity.Ok(namespaces)
+              cycle(result.index)
+            case Throw(e) =>
```

Comment:
I think we need to handle NotFounds here and turn them into an empty namespace list